Backport dev branch fix for favicons sometime not showing in
some consumers like qutebrowser. Not currently included in
upcoming 6.3.2 so may be needed beyond 6.3.1.

https://bugreports.qt.io/browse/QTBUG-103735
https://github.com/qt/qtwebengine/commit/c5de1b2123
--- a/src/core/favicon_driver_qt.cpp
+++ b/src/core/favicon_driver_qt.cpp
@@ -300,15 +300,15 @@ void FaviconDriverQt::DidStartNavigation(content::NavigationHandle *navigation_h
         return;
 
     m_faviconUrls.reset();
-    m_completedHandlersCount = 0;
-    m_latestFavicon = FaviconStatusQt();
 
     if (!navigation_handle->IsSameDocument()) {
+        m_completedHandlersCount = 0;
+        m_latestFavicon = FaviconStatusQt();
         m_documentOnLoadCompleted = false;
         m_manifestUrl = GURL();
-    }
 
-    m_viewClient->iconChanged(QUrl());
+        m_viewClient->iconChanged(QUrl());
+    }
 
     content::ReloadType reload_type = navigation_handle->GetReloadType();
     if (reload_type == content::ReloadType::NONE || IsOffTheRecord())
